src/odemis/util/transform.py

```diff
@@ -818,3 +818,176 @@ def inverse(self):
+        wcc = numpy.conj(w)  # w̄
+        w2 = w * w  # w²
+        wabs2 = w * wcc  # ww̄
+        wcc2 = wcc * wcc  # w̄²
+        w2wcc = w * wabs2  # w²w̄
+        wwcc2 = wcc * wabs2  # ww̄²
+        w3wcc2 = wabs2 * w2wcc  # w³w̄²
+        w2wcc3 = wabs2 * wwcc2  # w²w̄³
+        M = numpy.column_stack((numpy.ones_like(w),  # zero order
+                                w, wcc,  # first order
+                                w2, wabs2, wcc2,  # second order
+                                w2wcc, wwcc2,  # third order
+                                w3wcc2, w2wcc3))  # fifth order
+        return M
+
+    def __call__(self, x):
+        x = numpy.asarray(x)
+        w = x[..., 0] + 1.0j * x[..., 1]
+
+        M = self._vandermonde(w)
+        v = numpy.dot(M, self.coeffs)
+
+        if x.ndim == 1:
+            return np.array((v.real, v.imag))
+        return numpy.column_stack((v.real, v.imag))
+
+    @classmethod
+    def from_pointset(cls, x, y):
+        """
+        Estimate the transformation from a set of corresponding points.
+
+        Constructor for AnamorphosisTransform that determines the best
+        coordinate transformation from two point sets `x` and `y` in a
+        least-squares sense. For more information see [1].
+
+        Parameters
+        ----------
+        x : (n, 2) array
+            Coordinates in the source reference frame.
+        y : (n, 2) array
+            Coordinates in the destination reference frame. Must be of same
+            dimensions as `x`.
+
+        Returns
+        -------
+        tform : AnamorphosisTransform
+            Optimal coordinate transformation.
+
+        References
+        ----------
+        .. [1] J. Stopka, "Polynomial fit for Multi-beam distortions", internal
+        note, 2019
+
+        """
+        x = numpy.asarray(x)
+        y = numpy.asarray(y)
+
+        w = x[:, 0] + 1.0j * x[:, 1]
+        v = y[:, 0] + 1.0j * y[:, 1]
+
+        M = cls._vandermonde(w)
+        coeffs = numpy.linalg.lstsq(M, v)[0]
+        return cls(coeffs=coeffs)
+
+    @property
+    def coeffs(self):
+        tx, ty = self.translation
+        M = self.transformation_matrix
+        p = 0.5 * (M[0, 0] + M[1, 1])
+        q = 0.5 * (M[1, 0] - M[0, 1])
+        r = 0.5 * (M[0, 0] - M[1, 1])
+        s = 0.5 * (M[1, 0] + M[0, 1])
+        a = tx + 1.0j * ty
+        b1 = p + 1.0j * q
+        b2 = r + 1.0j * s
+        c1, c2, c3, d1, d2, e1, e2 = self.nlcoeffs
+        return (a, b1, b2, c1, c2, c3, d1, d2, e1, e2)
+
+    @coeffs.setter
+    def coeffs(self, coeffs):
+        a, b1, b2, c1, c2, c3, d1, d2, e1, e2 = coeffs
+        p, q = (b1.real, b1.imag)
+        r, s = (b2.real, b2.imag)
+        matrix = numpy.array([(p + r, -q + s), (q + s, p - r)])
+        self.transformation_matrix = matrix
+        self.nlcoeffs = (c1, c2, c3, d1, d2, e1, e2)
+        self.translation = (a.real, a.imag)
+
+    def inverse(self):
+        raise NotImplementedError()
```

Comment:
Is it possible to calculate the inverse? 

Could you add an explanation to the `NotImplementedError`?
`NotImplementedError("Not implemented because......")`

Reply:
OK; added message to the exception